diff/apply/apply.go

```diff
@@ -125,6 +125,8 @@ type readCounter struct {
 
 func (rc *readCounter) Read(p []byte) (n int, err error) {
 	n, err = rc.r.Read(p)
-	rc.c += int64(n)
+	if n > 0 {
```

Comment:
question: If the `rc.r.Read(p)` returns an error, the value of `n` should not matter right?, as `readCounter.Read()` will also be returning an error which the caller should recognise.

Reply:
> If the rc.r.Read(p) returns an error, the value of n should not matter right?

Matters, IIUC

https://pkg.go.dev/io#Reader

> When Read encounters an error or end-of-file condition after successfully reading n > 0 bytes, it returns the number of bytes read.

Reply:
Thank you